src/components/pages/CompanyData/components/CompanyAddressList.tsx

```diff
@@ -142,16 +152,11 @@ export const CompanyAddressList = ({
   }
 
   const getStatusColor = (status: string | undefined) => {
-    if (status === SharingStateStatusType.Success) {
-      return 'success'
-    } else if (
-      status === SharingStateStatusType.Pending ||
-      status === SharingStateStatusType.Initial
-    ) {
-      return 'warning'
-    } else {
-      return 'error'
-    }
+    if (status === SharingStateStatusType.Success) return 'success'
+    else if (status === SharingStateStatusType.Initial) return 'warning'
+    else if (status === SharingStateStatusType.Pending) return 'info'
+    else if (status === SharingStateStatusType.Ready) return 'primary'
+    return 'error'
```

Comment:
in discussion with Lavanya we came up with a solution using a map instead of a function, which is more elegant imho - can you please have a look and implement it accordingly:
https://github.com/eclipse-tractusx/portal-frontend/pull/967/files